`TestInterruptsDelay` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/9685351735/job/26725238503
#### Relevant snippet

```
=== FAIL: lib/player  (0.00s)
-test.shuffle 1719429247097369590
panic: test timed out after 10m0s
running tests:
	TestInterruptsDelay (10m0s)

goroutine 82 [running]:
testing.(*M).startAlarm.func1()
	/opt/go/src/testing/testing.go:2366 +0x265
created by time.goFunc
	/opt/go/src/time/sleep.go:177 +0x45

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc000131040, {0x36660ff, 0x13}, 0x37acd48)
	/opt/go/src/testing/testing.go:1750 +0x851
testing.runTests.func1(0xc000131040)
	/opt/go/src/testing/testing.go:2161 +0x86
testing.tRunner(0xc000131040, 0xc000b01b00)
	/opt/go/src/testing/testing.go:1689 +0x21f
testing.runTests(0xc0008e0258, {0x6605760, 0x7, 0x7}, {0x2?, 0xc000b01bf0?, 0x66302a0?})
	/opt/go/src/testing/testing.go:2159 +0x8bf
testing.(*M).Run(0xc00086d7c0)
	/opt/go/src/testing/testing.go:2027 +0xf18
main.main()
	_testmain.go:95 +0x2e5

goroutine 66 [chan receive, 10 minutes]:
github.com/gravitational/teleport/lib/player_test.TestInterruptsDelay(0xc0001311e0)
	/__w/teleport/teleport/lib/player/player_test.go:219 +0x599
testing.tRunner(0xc0001311e0, 0x37acd48)
	/opt/go/src/testing/testing.go:1689 +0x21f
created by testing.(*T).Run in goroutine 1
	/opt/go/src/testing/testing.go:1742 +0x826

goroutine 67 [select, 10 minutes]:
github.com/gravitational/teleport/lib/player.(*Player).applyDelay(0xc000585560, 0x0, 0x1388)
	/__w/teleport/teleport/lib/player/player.go:322 +0x32f
github.com/gravitational/teleport/lib/player.(*Player).stream(0xc000585560)
	/__w/teleport/teleport/lib/player/player.go:213 +0x6bc
created by github.com/gravitational/teleport/lib/player.New in goroutine 66
	/__w/teleport/teleport/lib/player/player.go:133 +0x845

goroutine 68 [select, 10 minutes]:
github.com/gravitational/teleport/lib/player_test.(*simpleStreamer).StreamSessionEvents.func1()
	/__w/teleport/teleport/lib/player/player_test.go:278 +0x4fd
created by github.com/gravitational/teleport/lib/player_test.(*simpleStreamer).StreamSessionEvents in goroutine 67
	/__w/teleport/teleport/lib/player/player_test.go:274 +0x147
FAIL	github.com/gravitational/teleport/lib/player	600.217s

=== FAIL: lib/player TestInterruptsDelay (unknown)
```

failed on master: https://github.com/gravitational/teleport/actions/runs/10167215810/job/28119097666

```
=== Failed
=== FAIL: lib/player  (0.00s)
-test.shuffle 1722362028531675143
panic: test timed out after 10m0s
running tests:
	TestInterruptsDelay (10m0s)

goroutine 50 [running]:
testing.(*M).startAlarm.func1()
	/opt/go/src/testing/testing.go:2366 +0x265
created by time.goFunc
	/opt/go/src/time/sleep.go:177 +0x45
...
```

on v16 backport: https://github.com/gravitational/teleport/actions/runs/11220253255/job/31188004797?pr=47246

```
=== FAIL: lib/player  (0.00s)
-test.shuffle 1728320451797324441
panic: test timed out after 10m0s
running tests:
	TestInterruptsDelay (10m0s)

goroutine 108 [running]:
testing.(*M).startAlarm.func1()
	/opt/go/src/testing/testing.go:2366 +0x265
created by time.goFunc
	/opt/go/src/time/sleep.go:177 +0x45

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc0009da4e0, {0x37a27e2, 0x13}, 0x38ed098)
	/opt/go/src/testing/testing.go:1750 +0x851
testing.runTests.func1(0xc0009da4e0)
	/opt/go/src/testing/testing.go:2161 +0x86
testing.tRunner(0xc0009da4e0, 0xc000b87b00)
	/opt/go/src/testing/testing.go:1689 +0x21f
testing.runTests(0xc0009a6ab0, {0x67fcb80, 0x8, 0x8}, {0x2?, 0xc000b87bf0?, 0x6827e40?})
	/opt/go/src/testing/testing.go:2159 +0x8bf
testing.(*M).Run(0xc000940fa0)
	/opt/go/src/testing/testing.go:2027 +0xf18
main.main()
	_testmain.go:97 +0x2e5
```

--- a/Makefile
+++ b/Makefile
@@ -915,7 +915,7 @@ test-go-prepare: ensure-webassets bpf-bytecode $(TEST_LOG_DIR) ensure-gotestsum 
 test-go-unit: FLAGS ?= -race -shuffle on
 test-go-unit: SUBJECT ?= $(shell go list ./... | grep -vE 'teleport/(e2e|integration|tool/tsh|integrations/operator|integrations/access|integrations/lib)')
 test-go-unit:
-	$(CGOFLAG) go test -cover -json -tags "$(PAM_TAG) $(FIPS_TAG) $(BPF_TAG) $(LIBFIDO2_TEST_TAG) $(TOUCHID_TAG) $(PIV_TEST_TAG) $(VNETDAEMON_TAG)" $(PACKAGES) $(SUBJECT) $(FLAGS) $(ADDFLAGS) \
+	$(CGOFLAG) GOEXPERIMENT=synctest go test -cover -json -tags "enablesynctest $(PAM_TAG) $(FIPS_TAG) $(BPF_TAG) $(LIBFIDO2_TEST_TAG) $(TOUCHID_TAG) $(PIV_TEST_TAG) $(VNETDAEMON_TAG)" $(PACKAGES) $(SUBJECT) $(FLAGS) $(ADDFLAGS) \
 		| tee $(TEST_LOG_DIR)/unit.json \
 		| gotestsum --raw-command -- cat
 

--- a/lib/player/player_experimental_test.go
+++ b/lib/player/player_experimental_test.go
@@ -1,0 +1,108 @@
+//go:build go1.24 && enablesynctest
+
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package player_test
+
+import (
+	"testing"
+	"testing/synctest"
+	"time"
+
+	"github.com/gravitational/teleport/lib/player"
+	"github.com/stretchr/testify/require"
+)
+
+/*
+This file uses the experimental testing/synctest package introduced with Go 1.24:
+
+    https://go.dev/blog/synctest
+
+When editing this file, you should set GOEXPERIMENT=synctest for your editor/LSP
+to ensure that the language server doesn't fail to recognize the package.
+
+This file is also protected by a build tag to ensure that `go test` doesn't fail
+for users who haven't set the environment variable.
+*/
+
+// TestInterruptsDelay tests that the player responds to playback
+// controls even when it is waiting to emit an event.
+func TestInterruptsDelay(t *testing.T) {
+	synctest.Run(func() {
+		p, err := player.New(&player.Config{
+			SessionID: "test-session",
+			Streamer:  &simpleStreamer{count: 3, delay: 5000},
+		})
+		require.NoError(t, err)
+
+		synctest.Wait()
+
+		require.NoError(t, p.Play())
+		t.Cleanup(func() { p.Close() })
+
+		synctest.Wait() // player is now waiting to emit event 0
+
+		// emulate the user seeking forward while the player is waiting..
+		start := time.Now()
+		p.SetPos(10_001 * time.Millisecond)
+
+		// expect event 0 and event 1 to be emitted right away
+		evt0 := <-p.C()
+		evt1 := <-p.C()
+		require.Equal(t, int64(0), evt0.GetIndex())
+		require.Equal(t, int64(1), evt1.GetIndex())
+
+		// Time will advance automatically in the synctest bubble.
+		// This assertion checks that the player was unblocked by
+		// the SetPos call and not because enough time elapsed.
+		require.Zero(t, time.Since(start))
+
+		<-p.C()
+
+		// the user seeked to 10.001 seconds, it should take 4.999
+		// seconds to get the third event that arrives at second 15.
+		require.Equal(t, time.Since(start), 4999*time.Millisecond)
+	})
+}
+
+func TestSeekForward(t *testing.T) {
+	synctest.Run(func() {
+		p, err := player.New(&player.Config{
+			SessionID: "test-session",
+			Streamer:  &simpleStreamer{count: 1, delay: 6000},
+		})
+		require.NoError(t, err)
+		t.Cleanup(func() { p.Close() })
+		require.NoError(t, p.Play())
+
+		start := time.Now()
+
+		time.Sleep(100 * time.Millisecond)
+		p.SetPos(500 * time.Millisecond)
+		time.Sleep(100 * time.Millisecond)
+		p.SetPos(5900 * time.Millisecond)
+
+		<-p.C()
+
+		// It should take 300ms for the event to be emitted.
+		// Two 100ms sleeps (200ms), then a seek to 5.900 seconds which
+		// requires another 100ms to wait for the event at 6s.
+		require.Equal(t, time.Since(start), 300*time.Millisecond)
+	})
+}

--- a/lib/player/player_test.go
+++ b/lib/player/player_test.go
@@ -173,57 +173,6 @@ func TestClose(t *testing.T) {
 	require.Equal(t, time.Second, p.LastPlayed())
 }
 
-func TestSeekForward(t *testing.T) {
-	clk := clockwork.NewRealClock()
-	p, err := player.New(&player.Config{
-		Clock:     clk,
-		SessionID: "test-session",
-		Streamer:  &simpleStreamer{count: 1, delay: 6000},
-	})
-	require.NoError(t, err)
-	t.Cleanup(func() { p.Close() })
-	require.NoError(t, p.Play())
-
-	time.Sleep(100 * time.Millisecond)
-	p.SetPos(500 * time.Millisecond)
-	time.Sleep(100 * time.Millisecond)
-	p.SetPos(5900 * time.Millisecond)
-
-	select {
-	case <-p.C():
-	case <-time.After(5 * time.Second):
-		require.FailNow(t, "event not emitted on time")
-	}
-}
-
-// TestInterruptsDelay tests that the player responds to playback
-// controls even when it is waiting to emit an event.
-func TestInterruptsDelay(t *testing.T) {
-	clk := clockwork.NewFakeClock()
-	p, err := player.New(&player.Config{
-		Clock:     clk,
-		SessionID: "test-session",
-		Streamer:  &simpleStreamer{count: 3, delay: 5000},
-	})
-	require.NoError(t, err)
-	require.NoError(t, p.Play())
-
-	t.Cleanup(func() { p.Close() })
-
-	clk.BlockUntil(1) // player is now waiting to emit event 0
-
-	// emulate the user seeking forward while the player is waiting..
-	p.SetPos(10_001 * time.Millisecond)
-
-	// expect event 0 and event 1 to be emitted right away
-	// even without advancing the clock
-	evt0 := <-p.C()
-	evt1 := <-p.C()
-
-	require.Equal(t, int64(0), evt0.GetIndex())
-	require.Equal(t, int64(1), evt1.GetIndex())
-}
-
 func TestRewind(t *testing.T) {
 	clk := clockwork.NewFakeClock()
 	p, err := player.New(&player.Config{

